src/Illuminate/Http/Request.php

```diff
@@ -139,16 +139,19 @@ public function fullUrl()
     /**
      * Get the full URL for the request with the added query string parameters.
      *
-     * @param  array  $query
+     * @param  array|null  $query The query string parameters to add to the URL
+     * @param  array|null  $only When specified, only the given query parameters will be returned
      * @return string
      */
-    public function fullUrlWithQuery(array $query)
+    public function fullUrlWithQuery(array $query = [], array $only = [])
     {
         $question = $this->getBaseUrl().$this->getPathInfo() === '/' ? '/?' : '?';
+        $query = array_merge($this->query(), $query);
+        $query = $only ? Arr::only($query, $only) : $query;
 
-        return count($this->query()) > 0
-            ? $this->url().$question.Arr::query(array_merge($this->query(), $query))
-            : $this->fullUrl().$question.Arr::query($query);
+        return count($query) > 0
```

Comment:
While you are at it, you could remove the unnecessary function call:
 ```suggestion
         return $query !== []
```

Reply:
I've kept the function call to maintain consistency with the other `fullUrlWithoutQuery` method.